src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/ConsoleLifetimeExitTests.cs

```diff
@@ -53,7 +64,16 @@ await Host.CreateDefaultBuilder()
             }
 
             // send the signal to the process
-            kill(remoteHandle.Process.Id, signal);
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // on Windows, we use the pipe to signal the process
```

Comment:
I am not sure how much is this simulation testing in practice. We had similar problem with testing PosixSignalRegistration on Windows and we just gave up on it there.

The proper way to test signals would be by having managed API to send signals to processes on Windows (https://github.com/dotnet/runtime/issues/109432 and related issues). Is this API on the list for .NET 10? Unfortunately, it would not help with this one since there does not appear to be a public Windows API to send CTRL_SHUTDOWN_EVENT.

Reply:
I feel there is value in this unit testing.  It does ensure that we allow thing to happen in the correct order and allow the process to exit cleanly while the background thread remains blocked. 

I did mention to Jeff that the process API would be helpful for *functional* testing here, but as you mention it is not sufficient for shutdown testing.  The best way to do that would be testing with containers.